src/components/date-time-picker/date-time-picker.spec.ts

```diff
@@ -0,0 +1,82 @@
+import { Component } from '@angular/core';
+import { async, ComponentFixture, TestBed } from '@angular/core/testing';
+import { FormsModule } from '@angular/forms';
+import { PopoverModule } from '../../components/popover';
+import { IconModule } from '../icon';
+import { DateTimePickerModule } from './date-time-picker.module';
+
+
+
+@Component({
+    selector: 'ux-components-date-time-picker',
+    template: `
+        <ux-date-time-picker
+            [(date)]="date"
+            [showTime]="showTime"
+            [showDate]="showDate"
+            [showTimezone]="showTimezone"
+            [showMeridian]="showMeridian"
+            [showSpinners]="showSpinners"
+            (dateChange)="onDateChange()">
+        </ux-date-time-picker>
+    `
+})
+export class DateRangePickerComponent {
+
+    onDateChange(): void { }
+
+    date: Date;
+    showTime: boolean = false;
+    showDate: boolean = true;
+    showTimezone: boolean = false;
+    showMeridian: boolean = true;
+    showSpinners: boolean = true;
+}
+
+describe('Date Time Picker', () => {
+    let component: DateRangePickerComponent;
+    let fixture: ComponentFixture<DateRangePickerComponent>;
+    let nativeElement: HTMLElement;
+
+    beforeEach(async(() => {
+        TestBed.configureTestingModule({
+            imports: [DateTimePickerModule, IconModule, PopoverModule, FormsModule],
+            declarations: [DateRangePickerComponent],
+        })
+            .compileComponents();
+    }));
+
+    beforeEach(() => {
+        fixture = TestBed.createComponent(DateRangePickerComponent);
+        component = fixture.componentInstance;
+        nativeElement = fixture.nativeElement;
+        fixture.detectChanges();
+    });
+
+    it('should initialise correctly', () => {
+        expect(component).toBeTruthy();
+    });
+
+    it('should update date and call onDateChange when start date is changed ', async () => {
+        spyOn(component, 'onDateChange');
+        component.date = new Date('Tue Jan 07 2020 00:00:00 GMT+0000 (Greenwich Mean Time)');
```

Comment:
Can I suggest changing this to use the alternate format for setting dates as it is less prone to mistakes:

```typescript
new Date(2020, 0, 7)
```